sdk/core/azure-core/src/main/java/com/azure/core/spatial/CollectionGeometry.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.spatial;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Represents a heterogeneous collection of {@link Geometry geometries}.
+ */
+public final class CollectionGeometry extends Geometry {
+    private final Collection<Geometry> geometries;
+
+    /**
+     * Constructs a geometry collection.
+     *
+     * @param geometries The geometries in the collection.
+     */
+    public CollectionGeometry(Collection<Geometry> geometries) {
```

Comment:
Collection isn't the most useful type in Java - should we do something more useful, such as List or Set?

Reply:
I'll change this to a list.